uploader/wiki_uploader.py

```diff
@@ -23,23 +23,49 @@ def __init__(
             access_secret=access_secret,
         )
 
-    def upload_file(self, file_name, file_stream, description=""):
+    def upload_file(self, file_name, file_stream, description="", license=""):
         if not description:
             description = file_name
 
         upload_result = self.mw_client.upload(
             file=file_stream,
             filename=file_name,
-            description=description,
+            description=get_initial_page_text(license, description),
             ignore=True,
-            comment="Uploaded with Google drive to commons.",
+            comment=description,
         )
         debug_information = "Uploaded: {0} to: {1}, more information: {2}".format(
             file_name, self.mw_client.host, upload_result
         )
         logging.debug(debug_information)
         upload_response = upload_result.get("result")
+        print(upload_response)
         if not upload_response == "Success":
             return False, {}
         else:
             return True, upload_result["imageinfo"]
+
+
+def get_initial_page_text(
+    license="", summary="", category="", date_of_creation="", source="", author=""
```

Comment:
since you have if checks, how about just 
```
license=None, summary=None, category=None, date_of_creation=None, source=None, author=None
```
and below:
```
source = "" if not source else  "|source={0}\n".format(source)
```

Reply:
(same for others)

Reply:
Makes sense. On it